examples/rpc_nats/main.go

```diff
@@ -0,0 +1,267 @@
+// On actual go-nitro service, only "A" related code would be present
+func nitroService(logger zerolog.Logger) {
+	// Setup logger
+	logger = logger.With().
+		Str("client", "NITRO ").
+		Str("scope", "     ").
+		Logger()
+
+	// logFile := "rpc_nats.log"
+	// truncateLog(logFile)
+	// logDestination := newLogWriter(logFile)
+
+	// TODO: refactor rpc service to allow chain and P2P MS updates
+	// for exemple: disconnect from B or I, reconnect to B or I, ...
+	// Orverall, the goal is to be able to completly control the client trough the rpc service
+
+	// Setup B and I clients
+	chainServiceB := chainservice.NewMockChainService(chain, bob.Address())
+	chainServiceI := chainservice.NewMockChainService(chain, irene.Address())
+
+	clientA, msgA := setupClientWithP2PMessageService(alice.PrivateKey, 3005, chainServiceA, logger)
+	clientB, msgB := setupClientWithP2PMessageService(bob.PrivateKey, 3006, chainServiceB, logger)
+	clientI, msgI := setupClientWithP2PMessageService(irene.PrivateKey, 3007, chainServiceI, logger)
+	peers := []p2pms.PeerInfo{
+		{Id: msgA.Id(), IpAddress: "127.0.0.1", Port: 3005, Address: alice.Address()},
+		{Id: msgB.Id(), IpAddress: "127.0.0.1", Port: 3006, Address: bob.Address()},
+		{Id: msgI.Id(), IpAddress: "127.0.0.1", Port: 3007, Address: irene.Address()},
+	}
+	println(clientA.ChainId)
+	// Connect nitro P2P message services
+	msgA.AddPeers(peers)
+	msgB.AddPeers(peers)
+	msgI.AddPeers(peers)
+
+	defer msgA.Close()
+	defer msgB.Close()
+	defer msgI.Close()
+
+	// Ignore B and I clients for now
+	_ = clientB
+	_ = clientI
+
+	// Setup A network service using transport from global variables (in global only because we currently use a mock transport)
+	trpA := nats2.NewNatsTransport(nc, []string{fmt.Sprintf("nitro.%s", network.DirectFundRequestMethod), "nitro.test-topic"})
+	conA, err := trpA.PollConnection()
+	if err != nil {
+		logger.Fatal().Msg(err.Error())
+	}
+	ntsA := network.NewNetworkService(conA, &serde.JsonRpc{})
+	ntsA.Logger = logger.With().Str("scope", "NETW ").Logger()
+
+	ntsA.RegisterRequestHandler(network.DirectFundRequestMethod, func(m *netproto.Message) {
+		if len(m.Args) < 1 {
+			logger.Fatal().Msg("unexpected empty args for direct funding method")
+			return
+		}
+
+		for i := 0; i < len(m.Args); i++ {
+			res := m.Args[i].(map[string]any)
+			directFundRequest := parser.ParseDirectFundRequest(res)
+
+			clientA.Engine.ObjectiveRequestsFromAPI <- directFundRequest
+		}
+	})
+
+	// TODO: complete example with B and I clients interactions (wait their own objectives, etc.)
+	ntsA.RegisterRequestHandler(network.DirectFundRequestMethod, func(m *netproto.Message) {
+		if len(m.Args) < 1 {
+			logger.Fatal().Msg("unexpected empty args for direct funding method")
+			return
+		}
+
+		for i := 0; i < len(m.Args); i++ {
+			res := m.Args[i].(map[string]any)
+			req := parser.ParseDirectFundRequest(res)
+
+			logger.Info().Msgf("Objective Request: %v", req)
+			clientB.Engine.ObjectiveRequestsFromAPI <- req
```

Comment:
It should be possible to reuse this method on `clientB`, and then we don't need to expose the `Engine` (`engine`):

https://github.com/statechannels/go-nitro/blob/0d46d6c5a5d97a2ad38e9229ae0725713301ea91/client/client.go#L133-L151